data/DataTrans.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+
+import pandas as pd
+import numpy as np
+import csv
+Max_Value=3000
+Min_Value=0
+
+def get_Zone(Z_csv):
+    zone=pd.read_csv(Z_csv)
+
+    # Create dictionary to store the mapping of states and codes
+    zone_dic={}
+    for i in range(len(zone)):
+        zone_dic[zone['STATE'][i]]=zone['ID'][i]
+    return zone_dic
+
+
+# Clean data; throw substations which are outside the United States and not available.
+def Clean(E_csv,zone_dic):
+    csv_data = pd.read_csv(E_csv)
+    Num_sub=len(csv_data)
+    row_indexs=[]
+    for i in range(Num_sub):
+        if((csv_data['STATE'][i] not in zone_dic) or (csv_data['STATUS'][i] != 'IN SERVICE') ):
+            row_indexs.append(i)
+    clean_data=csv_data.drop(labels=row_indexs)
+    return clean_data
+
+
+
+#Create dictionary to store neighbors of each sub
+def Neighbors(T_csv):
+    Trans_data = pd.read_csv(T_csv, usecols=['VOLTAGE','SUB_1','SUB_2'])
+    N_dict={}
+    for i in range(len(Trans_data)):
+        if(Trans_data['SUB_1'][i] in N_dict):
+            N_dict[Trans_data['SUB_1'][i]].append(Trans_data['SUB_2'][i])
+        else:
+            N_dict[Trans_data['SUB_1'][i]]=[Trans_data['SUB_2'][i]]
+        if(Trans_data['SUB_2'][i] in N_dict):
+            N_dict[Trans_data['SUB_2'][i]].append(Trans_data['SUB_1'][i])
+        else:
+            N_dict[Trans_data['SUB_2'][i]]=[Trans_data['SUB_1'][i]]
+    return N_dict
```

Comment:
Before we return the N_dict (which is a graph), it is probably a good idea to do island detection. 

Copied from a discussion I raise in our channel:

when I am debugging the data: there are quite some (~6000 out of 69637) substations inside Electric_Substations.csv that do not even show up in Electric_Power_Transmission_Lines.csv - in other words, no transmission lines connected to above substations and they are islands in the graph. It feels we should skip them… 

One example: filter POPLAR GAP in substation and it exists: https://hifld-geoplatform.opendata.arcgis.com/datasets/electric-substations/data?orderBy=TYPE&orderByAsc=false&where=NAME%20%3D%20%27POPLAR%20GAP%27. Filter this substation in either SUB1 or SUB2 returns nothing in transmission data: https://hifld-geoplatform.opendata.arcgis.com/datasets/electric-power-transmission-lines/data?orderBy=SUB_2&orderByAsc=false&where=SUB_1%20%3D%20%27POPLAR%20GAP%27
There are ~6000 such substations. (we already filter down to the “IN SERVICE” substation)